### DISCUSSION TITLE
Cannot get user wallet address from address variable into useWeb3Modal

### DISCUSSION THREAD

**Discussion by AudouxH** [WalletConnect/walletconnect-monorepo#2540] [Developer Support] [answered]
Hello everyone :hand: 

I'm currently working on a web3 connection page in react native, to get user wallet address and others information.
Link to the repository just here (https://github.com/AudouxH/React-native-wallet-connection).
 
I'm successfully integrated the web3Modal of walletConnect and i already open the modal and connect to trust wallet but when i'm redirect to my dapp, i cannot get the user wallet address from useWeb3Modal.

`const { isConnected, address, open, close, isOpen, provider } = useWeb3Modal();`

I got 2 error after being redirect to my dapp :
` {"context":"core"} {"context":"core/pairing"} {"context":"core/pairing/pairing"} No matching key, pairing undefined `
and
` {"context":"client"} [Error: No matching key, pairing undefined] `

I know that i'm connected and the modal got the user address because it's display in it but idk why it's not update into address variable.

I'll put you down some screenshot i took during the development, if you have any tips to give it to me, i'll take them :smile: 

![Error_from_pairing](https://github.com/WalletConnect/walletconnect-monorepo/assets/52534710/47fa2a10-657b-48ad-9a1f-be973ba03580)

![Error_1](https://github.com/WalletConnect/walletconnect-monorepo/assets/52534710/3bdb6c1a-f511-450f-883d-e2ee34182687)

![Error_2](https://github.com/WalletConnect/walletconnect-monorepo/assets/52534710/4d28570a-73bd-4bbf-9f03-8fc45302952f)

![Web3Modal_after_connection](https://github.com/WalletConnect/walletconnect-monorepo/assets/52534710/3a4fbdbd-f67a-49c7-86a9-353c9cd3230b)

Thank you per advance for anyone who can give me some advices or tips :+1:

**Comment by xzilja** [comment]
Which packages are you using?

**Reply by AudouxH** [reply]
Here is a list of my import i used for the project :

`import '@walletconnect/react-native-compat';`
`import { useWeb3Modal, Web3Modal } from '@web3modal/react-native';`

**Reply by xzilja** [reply]
@ignaciosantise Should be able to help here after the weekend 👍

**Comment by ignaciosantise** [comment, marked as answer]
hey @AudouxH 👋 `address` prop was added in [alpha.5 version](https://github.com/WalletConnect/web3modal-react-native/releases/tag/v1.0.0-alpha.5) and as i see [here](https://github.com/AudouxH/React-native-wallet-connection/blob/a344fd290d08df09f6f200259fd1bdedb1d249d6/package.json#LL16C48-L16C48) you are using an older version. Can you check if upgrading to the sdk latest version solves the issue?

Also, the `No matching key, pairing undefined` is unrelated to this

**Reply by AudouxH** [reply]
Hi @ignaciosantise ! I tried with the alpha.5 version and i finally got the wallet address perfectly thanks so much for the quick response ! :pray:
I still have the `No matching key, pairing undefined` error, do you have an idea from where that can come ?

**Reply by ignaciosantise** [reply]
Amazing! `No matching key, pairing undefined` is an issue from the wallet side, they need to update one of our libs, but the error doesn't affect functionalities

**Reply by AudouxH** [reply]
I also made Metamask connection today thanks to the package @Metamask/sdk, i'm so happy and gratefull, i should make party tonight :joy: ! Got it for the error, i will just wait for update :smile:. Thank you again for taking of your time to help me !
